packages/pieces/community/wedof/src/lib/actions/send-file.ts

```diff
@@ -82,18 +82,56 @@ export const sendFile = createAction({
         return fields;
       },
     }),
-    
-    file: Property.File({
-        displayName: "Fichier a envoyer",
-        required: true,
+    typeFile: Property.StaticDropdown({
+      displayName: "Choisir le format d'envoi du fichier",
+      description: "Permet de choisir la méthode d'envoi du fichier",
+      required: false,
+      defaultValue: "file",
+      options:  {
+          options: [
+            {
+              value: "file",
+              label: 'Attacher un document',
+            },
+            {
+              value: "url",
+              label: "Ajouter à partir d'un lien",
+            },
+          ],
+       disabled : false,
+      },
+      
     }),
-
+  files: Property.DynamicProperties({
+    description: '',
+    displayName: 'ez',
+    required: true,
+    refreshers: ['typeFile'],
+    props: async ({ typeFile }) => {
+      const _type = typeFile as unknown as string;
+      const props: DynamicPropsValue = {};
+      if (_type === "url") {
+        props['fileToDownload'] = Property.LongText({
+          displayName: "Lien vers le document",
+           description: 'URL du fichier à télécharger',
+          required: true,
+      });
+      } else if (_type === "file") {
+        props['file'] = Property.File({
+          displayName: "Fichier a envoyer",
+          required: true,
+      });
+      }
+      return props;
+    },
+  }),
   },
   async run(context) {
-    const message = {
+     const message = {
         title: context.propsValue.title ?? null,
-        typeId: context.propsValue.typeId,
-        file: context.propsValue.file,
+        typeId: String(context.propsValue.typeId['undefined'])?? null,
```

Comment:
@obenazouz-wedof, could you please recheck the typeId prop? It appears to be accessing an 'undefined' field.